### ISSUE TITLE
[a11y]: How to set the aria-label of an empty option? 

### ISSUE THREAD

**Issue by gcattan** [carbon-design-system/carbon#10998]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

v7.55.1

### Automated testing tool and ruleset

IBM Accessibility Tester

### Description

I have a combobox with an empty option.
I would like this option to stay empty and have a `aria-label` equals to `Default option`:

![image](https://user-images.githubusercontent.com/6229031/158560468-3d788981-735d-454e-8806-abd098147a11.png)

I do not see how I can do it programmatically.

### CodeSandbox example

https://codesandbox.io/s/bold-thunder-55q3w2?file=/src/index.js

### Steps to reproduce

1. Create an Combobox with an empty option
2. Open the dropdown 
3. Start IBM Accessibility Checker

![image](https://user-images.githubusercontent.com/6229031/158559483-7c93ab72-61e9-43ae-8c29-99cb149c8e79.png)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by dakahn** [comment]
the error comes from the aria label being programmatically derived from the list items name. what is the use case for a blank option in a drop down list?

**Comment by gcattan** [comment]
Here is the business justification left by my predecessor: 

```
 add in a blank option so the user can deselect the current value if:
- field is not required, or
- field has no selected value (the blank is the current selection)
```

**Comment by tay1orjones** [comment]
@gcattan does the "x to clear" functionality not work for both those use cases?


https://user-images.githubusercontent.com/3360588/159054299-68ecba04-8505-45bf-8cca-19f663f1b2d3.mov

**Comment by tay1orjones** [comment]
@gcattan I dug into this a bit and found we've addressed this before in https://github.com/carbon-design-system/carbon/issues/6581

In summary, the intended spec is for the first option to be `[Select an option]`, which would avoid the accessibility error you've ran into with using a blank option.

The issue referenced above contains a [code example](https://codesandbox.io/s/optimistic-glitter-mpizz?file=/src/index.js) as well.

**Comment by gcattan** [comment]
Thank you for digging into it. 
However, if the input is enabled, we do not want the user to be able to edit a `[Select an option]` string. And in some cases we want our default option to be empty.

We could achieve this using the `renderItemToElement` in order to render a label different from our item, but IMO this functionality should be built-in into the component itself, as this behavior is not consistent between Combobox and dropdown.

**Comment by tay1orjones** [comment]
@gcattan Sorry for the delay here. Dropdown and Combobox are intended to support two different (but similar) interaction patterns. Differing behavior between Combobox and Dropdown is intentional to meet the needs these differing interaction patterns.

The designed UX specification for these two as optional elements is the following:

- Dropdown: provide a `[Select an option]` option
- Combobox: use the `x to clear` function

Right now these components are not designed to contain the same list of options. To consider a change we'd need more info as to why the above does not meet your use case. cc @carbon-design-system/design

**Comment by gcattan** [comment]
Hi, it is been a long time, and to be honest I do not remember the details here. We implemented the workaround with the `renderItemToElement` which somehow answers my initial question. Why we were not able to use directly the `x to clear` function is probably a different issue.
